diktat-cli/src/main/kotlin/org/cqfn/diktat/cli/DiktatProperties.kt

```diff
@@ -0,0 +1,242 @@
+package org.cqfn.diktat.cli
+
+import org.cqfn.diktat.DiktatRunnerArguments
+import org.cqfn.diktat.api.DiktatProcessorListener
+import org.cqfn.diktat.api.DiktatReporterFactory
+import org.cqfn.diktat.common.config.rules.DIKTAT
+import org.cqfn.diktat.common.config.rules.DIKTAT_ANALYSIS_CONF
+import org.cqfn.diktat.util.isKotlinCodeOrScript
+import org.cqfn.diktat.util.tryToPathIfExists
+import org.cqfn.diktat.util.walkByGlob
+import generated.DIKTAT_VERSION
+import generated.KTLINT_VERSION
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.core.LoggerContext
+import org.slf4j.event.Level
+import java.io.OutputStream
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.createDirectories
+import kotlin.io.path.outputStream
+import kotlin.system.exitProcess
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ArgType
+import kotlinx.cli.default
+import kotlinx.cli.vararg
+
+/**
+ * @property config path to `diktat-analysis.yml`
+ * @property mode mode of `diktat`
+ * @property reporterProviderId
+ * @property output
+ * @property patterns
+ */
+data class DiktatProperties(
+    val config: String,
+    val mode: DiktatMode,
+    val reporterProviderId: String,
+    val output: String?,
+    private val groupByFileInPlain: Boolean,
+    private val colorNameInPlain: String?,
+    private val logLevel: Level,
+    val patterns: List<String>,
+) {
+    /**
+     * Configure logger level using [logLevel]
+     */
+    fun configureLogger() {
+        // set log level
+        LogManager.getContext(false)
+            .let { it as LoggerContext }
+            .also { ctx ->
+                ctx.configuration.rootLogger.level = when (logLevel) {
+                    Level.ERROR -> org.apache.logging.log4j.Level.ERROR
```

Comment:
Up to you, but you can 

```kotlin
import org.apache.logging.log4j.Level as Log4jLevel
```

or even 

```kotlin
import org.apache.logging.log4j.Level.ERROR as LOG4J_ERROR
```

at the beginning of your file.

Feel free to leave w/o resolution, though.

Reply:
it's used at in a single place. will leave as is